src/ui/resource/Image.js

```diff
@@ -419,9 +419,9 @@ exports = Class(lib.PubSub, function () {
 	};
 
 	this._renderImage = function(ctx, srcImg, srcX, srcY, srcW, srcH, destX, destY, destW, destH) {
```

Comment:
We should add protections around `drawImage` calls to ensure valid parameters that don't cause errors.

I separated `_renderImage` from `render` for the sole purpose of isolating the try-catch block so that `render` could be optimized. Since we're removing the try-catch, I see two options here. We get rid of `_renderImage` or we use it to house all of the parameter validation for `drawImage` ... it could even be made "public" and hypothetically used in other ways with the insurance of safe calls even with broken params.